lib/pure/typetraits.nim

```diff
@@ -183,3 +183,32 @@ since (1, 1):
 
     type T2 = T
     genericParamsImpl(T2)
+
+func rangeof*(T: typedesc): Slice[T] {.inline, since: (1, 3, 5).} =
```

Comment:
This should be named ``sliceof`` and then it's ready to go as far as I'm concerned.

Reply:
`sliceof` sounds weird IMO. Example:

```nim
assert 42 in sliceof(Positive):
```

It's hard to understand that statement without knowing what slices are in Nim, and even then you'd need to whip out the docs just to know what `sliceof()` does.

Reply:
Well it returns a ``Slice[T]`` so I have to know Nim's slice type.

Reply:
Yes, but arguably, comparing between these:

```nim
assert 42 in rangeof(Positive)

assert 42 in sliceof(Positive)
```

The former does a better job at saying "I want to make sure that 42 is within the range of type `Positive`".

You can know `Slice[T]` and still have to stop to look at the docs because `sliceof()` just sound alien.